app/main.py

```diff
@@ -1,3 +1,67 @@
+from __future__ import annotations
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> Distance:
+        if isinstance(other, int | float):
+            km = self.km + other
+        else:
+            km = self.km + other.km
```

Comment:
Other can be `Any` type, This means, that `other` can be str, list, etc. So, you must improve your type hints for `other` variables.